proxy/client/storage/storage.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018-2023 contributors to the OpenLineage project
+// SPDX-License-Identifier: Apache-2.0
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/OpenLineage/OpenLineage/client-proxy/config"
+	_ "github.com/mattn/go-sqlite3"
+	"math/rand"
+	"time"
+)
+
+type Storage struct {
+	db *sql.DB
+}
+
+type Partition struct {
+	RowID     uint64
+	Name      string
+	Size      uint64
+	CreatedAt time.Time
+	IsCurrent bool
+}
+
+// IStorage is an interface for interaction with local storage database
+type IStorage interface {
+	GetCurrentPartition() (*Partition, error)
+	RotatePartition() (*Partition, error)
+	InsertLineageEvent(*Partition, string) error
```

Comment:
_Minor_: I would be consistent with method names (_wordy_ vs _less_). That is, in `EventLogger` you define:

```
EventLogger.Log()
```

And for the storage you have:

```
IStorage.InsertLineageEvent()
```

I think a method naming strategy would help keep things consistent. I would either be _wordy_ with `EventLogger.LogLineageEvent()` or _less_ and go with `IStorage.Insert()`